src/container-mapping.ts

```diff
@@ -0,0 +1,184 @@
+    */
+    private async _runPostJob() {
+        let startTime = core.getState('PreJobStartTime');
+        if (startTime.length <= 0) {
+            startTime = new Date(new Date().getTime() - 10000).toISOString();
+            console.log(`PreJobStartTime not defined, using now-10secs`);
+        }
+        console.log(`PreJobStartTime: ${startTime}`);
+
+        let reportData = {
+            dockerVer: "",
+            dockerEvents: "",
+            dockerImages: ""
+        };
+        
+        // Initialize the commands 
+        await exec.exec('docker --version', null, {
+            listeners: {
+                stdout: (data: Buffer) => {
+                    reportData.dockerVer = reportData.dockerVer.concat(data.toString());
+                }
+            }
+        });
+        await exec.exec(`docker events --since ${startTime} --until ${new Date().toISOString()} --filter event=push --filter type=image --format ID={{.ID}}`, null, {
+            listeners: {
+                stdout: (data: Buffer) => {
+                    reportData.dockerEvents = reportData.dockerEvents.concat(data.toString());
+                }
+            }
+        });
+        await exec.exec('docker images --format CreatedAt={{.CreatedAt}}::Repo={{.Repository}}::Tag={{.Tag}}::Digest={{.Digest}}', null, {
+            listeners: {
+                stdout: (data: Buffer) => {
+                    reportData.dockerImages = reportData.dockerImages.concat(data.toString());
+                }
+            }
+        });
+
+        core.debug("Finished data collection, starting API calls.");
+
+        await this.sendReport(reportData, sendReportRetryCount);
+    }
+
+    /**
+     * Sends a report to Defender for DevOps and retries on the specified count
+     * @param data the data to send
+     * @param retryCount the number of time to retry
+     * @returns a Promise
+     */
+    private async sendReport(data: Object, retryCount: number = 0): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            do {
+                try {
+                    await this._sendReport(data);
+                    resolve();
+                    break;
+                } catch (error) {
+                    if (retryCount == 0) {
+                        reject('Failed to send report: ' + error);
+                    } else {
+                        retryCount--;
+                        core.debug(`Retrying API call. Retry count: ${retryCount}`);
+                    }
+                }
+            } while (retryCount >= 0)
+        });
+    }
+    
+    /**
+     * Sends a report to Defender for DevOps
+     * @param data the data to send
+     * @returns a Promise
+     */
+    private async _sendReport(data: Object): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            let apiTime = new Date().getMilliseconds();
+            var bearerToken = await core.getIDToken();

```

Comment:
typescript does not do super well with nested promise failures, so if this throws, the whole program might fail with `unhandledException` or something. One easy way to handle it is to just chain this here - 

```typescript
await core.getIDToken()
.catch(error => {
            reject(`Failed to get ID token: ${error}`);
        });
```